src/main/daml/Daml/Finance/Util/Disclosure.daml

```diff
@@ -0,0 +1,101 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Util.Disclosure where
+
+import DA.Foldable qualified as F (mapA_)
+import DA.Map qualified as M (delete, insert, lookup)
+import DA.Optional (fromOptional)
+import DA.Set (Set)
+import DA.Set qualified as S (difference, empty, intersection, null, union)
+import DA.Traversable qualified as T (mapA)
+import Daml.Finance.Interface.Types.Common.Types (PartiesMap)
+import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (AddObservers(..), I, RemoveObservers(..), SetObservers(..))
+
+setObserversImpl :
```

Comment:
Could we add a description for these functions? They are used throughout the entire library as default implementations